lib/URI.pm

```diff
@@ -100,8 +114,16 @@ sub _init
 sub _fix_uric_escape_for_host_part {
   return if HAS_RESERVED_SQUARE_BRACKETS;
   return if $_[0] !~ /%/;
+  return if $_[0] =~ m,^(?:$URI::schemes_without_host_part_re):,os;
+
+  # until a scheme specific handler is available, fall back to previous behavior of v5.10 (i.e. 'mailto:')
+  if ($_[0] =~ m,^(?:$URI::fallback_schemes_re):,os) {
+    $_[0]    =~ s/\%5B/[/gi;
+    $_[0]    =~ s/\%5D/]/gi;
+    return;
+  }
 
-  if ($_[0] =~ m,^((?:$URI::scheme_re:)?)//([^/?\#]*)(.*)$,os) {
+  if ($_[0] =~ m,^((?:$URI::scheme_re:)?)//([^/?\#]+)(.*)$,os) {
     my $orig          = $2;
```

Comment:
Also curl brace delimiters here?

Reply:
Absolutely. This §$&(/%§ line caused all the trouble.